be/src/column/column_access_path.cpp

```diff
@@ -136,4 +136,139 @@ const std::string ColumnAccessPath::to_string() const {
     return ss.str();
 }
 
+void ColumnAccessPathUtil::rewrite_struct_type_descriptor(TypeDescriptor& original_type,
+                                                          const ColumnAccessPathPtr& access_path) {
+    DCHECK(original_type.is_struct_type());
+
+    if (is_select_all_subfields(access_path)) {
+        return;
+    }
+
+    // build child access path mapping
+    std::unordered_map<std::string, const ColumnAccessPathPtr&> subfield_mapping{};
+    for (const auto& child : access_path->children()) {
+        subfield_mapping.emplace(child->path(), child);
+    }
+
+    std::vector<std::string> new_subfield_names{};
+    std::vector<TypeDescriptor> new_subfield_types{};
+
+    for (size_t i = 0; i < original_type.children.size(); i++) {
+        TypeDescriptor& subfield_type = original_type.children[i];
+        const auto& subfield_name = original_type.field_names[i];
+
+        const auto& it = subfield_mapping.find(subfield_name);
+        if (it == subfield_mapping.end()) {
+            continue;
+        }
+
+        if (subfield_type.is_complex_type()) {
+            const ColumnAccessPathPtr& child_path = it->second;
+            DCHECK(child_path->is_field());
+            rewrite_complex_type_descriptor(subfield_type, child_path);
+        }
+
+        new_subfield_names.emplace_back(subfield_name);
+        new_subfield_types.emplace_back(subfield_type);
+    }
+
+    original_type.children = new_subfield_types;
+    original_type.field_names = new_subfield_names;
```

Comment:
```suggestion
    original_type.children = std::move(new_subfield_types);
    original_type.field_names = std::move(new_subfield_names);
```
Use move assignment to avoid unnecessary copy.